packages/components/list-item/src/list-item.js

```diff
@@ -37,13 +42,24 @@ customElementDefineHelper(
 			return colorType;
 		}
 
+		slotChangeHandler(e) {
+			this.hasSlottedIcon = true;
+			const slottedNodes = e.currentTarget.assignedNodes();
+			validateSlottedNodes(this.tagName, slottedNodes, ['TS-APP-ICON']);
+			this.requestUpdate();
+		}
+
 		get iconLeftTemplate() {
 			const icon = this.icon || this.iconLeft;
-			if (icon) {
-				return html`
-					<ts-icon class="icon-left" icon="${icon}" size="large" type="${this.colorType}"></ts-icon>
-				`;
-			}
+			return icon
+				? html`
+						<ts-icon class="icon-left" icon="${icon}" size="large" type="${this.colorType}"></ts-icon>
+				  `
+				: html`
+						<span class="${this.hasSlottedIcon ? 'icon-left' : ''}">
+							<slot @slotchange="${this.slotChangeHandler}"></slot>
```

Comment:
maybe using named slot make it more extensible for future, if we need another slot.